<launch>
  <param name="use_sim_time" value="false"/>

	<node pkg="rviz" type="rviz" name="rviz" 
	  args="-d $(find ar_pose)/launch/live_single.vcg"/>
		    
   <node pkg="tf" type="static_transform_publisher" name="world_to_cam" 
     args="1 1 0.3 0 0 0 world ar_marker 10" />

	<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" respawn="false" output="log">
		<param name="video_device" type="string" value="/dev/video0"/>
		<param name="camera_frame_id" type="string" value="usb_cam"/>
		<param name="io_method" type="string" value="mmap"/>
		<param name="image_width" type="int" value="640"/>
		<param name="image_height" type="int" value="480"/>
		<param name="pixel_format" type="string" value="mjpeg"/>
		<rosparam param="D">[265.000000, 257.000000, 4.600000, 1.002303, 0.0000]</rosparam>
		<rosparam param="K">[558.70655574536931, 0.0, 316.68428342491319, 0.0, 553.44501004322387, 238.23867473419315, 0.0, 0.0, 1.0]</rosparam>
		<rosparam param="R">[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]</rosparam>
		<rosparam param="P">[558.70655574536931, 0.0, 316.68428342491319, 0.0, 0.0, 553.44501004322387, 238.23867473419315, 0.0, 0.0, 0.0, 1.0, 0.0]</rosparam>
   </node>
	
	<node name="ar_pose" pkg="ar_pose" type="ar_single" respawn="false" output="screen">
		<param name="marker_pattern" type="string" value="data/patt.hiro"/>
		<param name="marker_width" type="double" value="80.0"/>
		<param name="marker_center_x" type="double" value="0.0"/>
		<param name="marker_center_y" type="double" value="0.0"/>
		<param name="threshold" type="int" value="100"/>
		<param name="use_history" type="bool" value="true"/>
	</node>
</launch>
